runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -392,4 +408,59 @@ public State waitUntilFinish(Duration duration) throws IOException {
           "DirectPipelineResult does not support waitUntilFinish.");
     }
   }
+
+  /**
+   * Internal configuration options for the {@link DirectRunner}. Deals with internal implementation
+   * details.
+   */
+  public interface InternalOptions extends DirectOptions {
+    /**
+     * Gets the {@link ExecutorServiceFactory} to use to create instances of {@link ExecutorService}
+     * to execute {@link PTransform PTransforms}.
+     *
+     * <p>Defaults to a {@link FixedThreadPoolSupplier}, which produces instances of
+     * {@link Executors#newFixedThreadPool(int)}.
+     */
+    @JsonIgnore
+    @Required
+    @Hidden
+    @Default.InstanceFactory(FixedThreadPoolSupplier.class)
+    OptionsSupplier<ExecutorService> getExecutorServiceFactory();
+
+    void setExecutorServiceFactory(OptionsSupplier<ExecutorService> executorService);
+
+    /**
+     * Gets the {@link Clock} used by this pipeline. The clock is used in place of accessing the
+     * system time when time values are required by the evaluator.
+     */
+    @Default.InstanceFactory(NanosOffsetClock.Factory.class)
```

Comment:
Done. I forgot that the Pipeline provides a getRunner method that provides the actual runner.